R/variable-derivation.R

```diff
@@ -0,0 +1,82 @@
+#' Get or set a derived variable's `CrunchExpr`
+#'
+#' Get a derived variable's derivation formula as a (`CrunchExpr`)[expressions] with
+#' `derivation([variable])`. Set (change) a derived variable's derivation with
+#' `derivation([variable]) <- [expression]`
+#'
+#' @param x a variable
+#' @param value a `CrunchExpr` to be used as the derivation (for the setter only)
+#'
+#' @return a `CrunchExpr` of the derivation
+#'
+#' @examples
+#' \dontrun{
+#'
+#' ds$derived_v1 <- ds$v1 + 5
+#'
+#' derivation(ds$derived_v1)
+#' # Crunch expression: v1 + 5
+#'
+#' derivation(ds$derived_v1) <- ds$v1 + 10
+#' derivation(ds$derived_v1)
+#' # Crunch expression: v1 + 10
+#' }
+#' @name derivations
+#' @aliases derivation derivation<-
+NULL
+
+#' @export
+#' @rdname derivations
+setMethod("derivation", "CrunchVariable", function(x) {
+    if (!is.derived(x)) {
+        return(NULL)
+    }
+    expr <- CrunchExpr(expression=entity(x)@body$derivation)
+    expr@expression <- absolutifyVariables(expr@expression, x)
+
+    return(expr)
+})
+
+absolutifyVariables <- function (expr, var) {
+    if ("function" %in% names(expr)) {
+        expr[["args"]] <- absolutifyExpressionArgs(expr[["args"]], var)
```

Comment:
With a few slight modifications, yeah that works and is much cleaner. Thanks! I was relying a bit too much on the pattern I had seen for other recursive functions like this.